_python/main/models.py

```diff
@@ -2620,6 +2630,19 @@ def set_password(self, raw_password):
 AnonymousUser.is_superadmin = False
 
 
+def update_user_login_fields(sender, request, user, **kwargs):
+    """
+        Register signal to record user login details on successful login.
+    """
+    user.last_login_at = user.current_login_at
+    user.current_login_at = timezone.now()
+    user.last_login_ip = user.current_login_ip
+    user.current_login_ip = get_ip_address(request)
+    user.login_count += 1
+    user.save(update_fields=['last_login_at', 'current_login_at', 'last_login_ip', 'current_login_ip', 'login_count'])
+user_logged_in.connect(update_user_login_fields)
```

Comment:
Nice idea to use a signal. I wonder if this level of tracking is handy, or is something we might drop going forward. Since you've done all the work, I suppose no reason to drop it.

Reply:
For documentation -- the default Django behavior is to use this same signal to set `last_login = timezone.now()`, so we could fully switch to the Django way by renaming current_login_at to last_login and dropping the signal and the other fields. I can imagine situations where more logging is handy though. 🤷‍♂️